File: C++/264_UglyNumberII.cpp
class Solution {
public:
    int nthUglyNumber(int n) {
        int two = 0, three = 0, five = 0;
        int nextTwo = 2, nextThree = 3, nextFive = 5;
        int minny = 0;
        vector<int> ugly = {1};
        while(ugly.size() < n)
        {
            minny = min({nextTwo,nextThree,nextFive});
            ugly.push_back(minny);
            if (nextTwo == ugly[ugly.size()-1])
            {
                two++;
                nextTwo = ugly[two] * 2;
            }
            if (nextThree == ugly[ugly.size()-1])
            {
                three++;
                nextThree = ugly[three] * 3;
            }
            if (nextFive == ugly[ugly.size()-1])
            {
                five++;
                nextFive = ugly[five] * 5;
            }
        }
        return ugly[ugly.size()-1];
    }
};

File: 264_UglyNumberII.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        two = three = five = 0
        nextTwo = 2
        nextThree = 3
        nextFive = 5
        ugly = [0] * n
        ugly[0] = 1
        for i in range(1,n):
            ugly[i] = min( nextTwo, nextThree, nextFive)
            if nextTwo == ugly[i]: 
                two += 1 
                nextTwo = ugly[two]*2
            if nextThree == ugly[i]: 
                three+= 1
                nextThree = ugly[three]*3
            if nextFive == ugly[i]: 
                five+= 1
                nextFive = ugly[five]*5
        return ugly[-1]